horizon_bsn/bsndashboard/static/neutron.service.js

```diff
@@ -0,0 +1,338 @@
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+(function () {
+  'use strict';
+
+  angular
+    .module('horizon.app.core.openstack-service-api')
+    .factory('horizon.app.core.openstack-service-api.bsnneutron', neutronAPI);
+
+  neutronAPI.$inject = [
+    'horizon.framework.util.http.service',
+    'horizon.framework.widgets.toast.service'
+  ];
+
+  /**
+   * @ngdoc service
+   * @name neutronAPI
+   * @param {Object} apiService
+   * @param {Object} toastService
+   * @description Provides access to Neutron APIs.
+   * @returns {Object} The service
+   */
+  function neutronAPI(apiService, toastService) {
+    var service = {
+      reachabilitytest_create: reachabilitytest_create,
+      reachabilitytest_list: reachabilitytest_list,
+      reachabilitytest_run: reachabilitytest_run,
+      reachabilitytest_delete: reachabilitytest_delete,
+
+      networktemplate_create: networktemplate_create,
+      networktemplate_list: networktemplate_list,
+      networktemplate_get: networktemplate_get,
+      networktemplate_update: networktemplate_update,
+      networktemplate_delete: networktemplate_delete,
+
+      heatstack_delete: heatstack_delete,
+      heatstack_create: heatstack_create,
+      check_status: check_status,
+      template_validate: template_validate,
+      networktemplateassignment_create: networktemplateassignment_create,
+      networktemplateassignment_list: networktemplateassignment_list,
+      networktemplateassignment_update: networktemplateassignment_update,
+      networktemplateassignment_delete: networktemplateassignment_delete,
+      
+      reachabilityquicktest_create: reachabilityquicktest_create,
+      reachabilityquicktest_get: reachabilityquicktest_get,
+      reachabilityquicktest_update: reachabilityquicktest_update,
+
+      router_get: router_get,
+      router_update: router_update
+    };
+
+    return service;
+
+    /////////////
+
+    // Neutron Services
+
+
+    /**
+     * //////////////////////////////////////////
+     * REACHABILITY TESTS
+     * //////////////////////////////////////////
+     */
+    /**
+     * @name reachabilitytest_create
+     * @param {Object} test - The test
+     * @description Create a reachability test.
+     *
+     * @returns {Object} The result of the creation call.
+     */
+    function reachabilitytest_create(test) {
+      return apiService.post('api/neutron/reachabilitytests/', test)
+        .error(function () {
+          toastService.add('error', gettext('Error creating reachability test'));
```

Comment:
this is good!
in addition, can you check if its possible to also print the error received from the API? neutron would say an error as `test with that name already exists`. so does that come in the error message or just the static message listed above?
try it out and let me know.